### PR TITLE
NostoProvider: Extract client script loading functionality to a hook

### PR DESCRIPTION
**Pull request by nosto-mikpou**
Extracts client script functionality to a useClientScriptLoad hook + minor cleanup

### CODE REVIEW

File: src/hooks/useClientScriptLoad.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    if (shopifyMarkets) {
+      const existingScript = document.querySelector("[nosto-client-script]")
+      const nostoSandbox = document.querySelector("#nosto-sandbox")
+      
+      const marketId = String(shopifyMarkets?.marketId || "")
+      const language = shopifyMarkets?.language || ""
+
+      if (
+        !existingScript ||
+        existingScript?.getAttribute("nosto-language") !== language ||
```
Comment (nosto-mikpou) [inline src/hooks/useClientScriptLoad.ts:59]: Is the language check needed?

File: src/hooks/useClientScriptLoad.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      window.nostojs = (cb: (api: NostoClient) => void) => {
+        (window.nostojs.q = window.nostojs.q || []).push(cb)
+      }
+      window.nostojs(api => api.setAutoLoad(false))
+    }
+
+    // Load Nosto client script if not already loaded externally
+    if (!isNostoLoaded() && !shopifyMarkets) {
+      const urlPartial = `/include/${account}`
+      const script = createScriptElement(urlPartial)
```
Comment (manikandan-ravikumar) [inline src/hooks/useClientScriptLoad.ts:44]: Here, can we have a check if the client script is already loaded on page and skip this if possible? Or else we could introduce a config flag through which we could skip adding duplicate script tags

Comment (manikandan-ravikumar) [reply src/hooks/useClientScriptLoad.ts:44]: In a separate PR also fine

File: src/hooks/useClientScriptLoad.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import { useState, useMemo, useEffect } from "react"
+import { isNostoLoaded } from "../components/helpers"
+import type { NostoClient } from "../types"
+import type { NostoProviderProps } from "../components/NostoProvider"
+
+type NostoScriptProps = Pick<NostoProviderProps, "account" | "host" | "shopifyMarkets">
+
+export function useClientScriptLoad(props: NostoScriptProps) {
```
Comment (manikandan-ravikumar) [inline src/hooks/useClientScriptLoad.ts:8]: rename this to `useLoadClientScript` or `useClientScriptLoader`

File: src/hooks/useClientScriptLoad.ts (hunk 2 above)
Comment (nosto-mikpou) [reply src/hooks/useClientScriptLoad.ts:44]: Sure, we have loadScript flag in the [WIP PR](https://github.com/Nosto/nosto-react/pull/95/files#diff-1e68849bba5d7e4b3a74f844d6bc8d96604019ab29ae0267ef2b93db101ac5cdR44) waiting to be transferred in the next PR.

Comment (timowestnosto) [reply src/hooks/useClientScriptLoad.ts:44]: there is already a loaded check (isNostoLoaded) and the flag will be introduced in a later PR

**Review by manikandan-ravikumar — APPROVED** [review]

File: src/hooks/useClientScriptLoad.ts (hunk 3 above)
Comment (timowestnosto) [reply src/hooks/useClientScriptLoad.ts]: useLoadClientScript would be my preference

File: src/hooks/useClientScriptLoad.ts (hunk 1 above)
Comment (manikandan-ravikumar) [reply src/hooks/useClientScriptLoad.ts]: May need it for markets. I can't say for sure though

**Review by timowestnosto — APPROVED** [review]

File: src/hooks/useClientScriptLoad.ts (hunk 1 above)
Comment (timowestnosto) [reply src/hooks/useClientScriptLoad.ts]: this is a language mismatch check, it makes sense

**Pull request merged**